File: packages/daily/web/src/application/ports/dto/TagSet.ts
export interface TagSetGroupDTO {
  readonly id: string;
  readonly name: string;
  readonly weight: number;
  readonly created_at: string;
  readonly updated_at: string;
}

export interface TagSetDTO {
  readonly id: string;
  readonly name: string;
  readonly group_id: string | null;
  readonly tags_any: string[];
  readonly tags_all: string[];
  readonly tags_exclude: string[];
  readonly weight: number;
  readonly last_used_at: string | null;
  readonly created_at: string;
  readonly updated_at: string;
}

File: packages/daily/server/internal/domain/entity/tag_set.go
package entity

import "time"

type TagSetGroup struct {
	ID        string
	UserID    int64
	Name      string
	Weight    int
	CreatedAt time.Time
	UpdatedAt time.Time
}

type TagSet struct {
	ID          string
	UserID      int64
	GroupID     *string
	Name        string
	TagsAny     string // JSON array
	TagsAll     string // JSON array
	TagsExclude string // JSON array
	Weight      int
	LastUsedAt  *time.Time
	CreatedAt   time.Time
	UpdatedAt   time.Time
}
